### PR TITLE
STOR-1287: KCMO and MCO must set env var for CSI migration

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/STOR-1287
/cc @openshift/storage

- [STOR-1267](https://issues.redhat.com//browse/STOR-1267): Bump openshift/api
- [STOR-1287](https://issues.redhat.com//browse/STOR-1287): KCMO and MCO must set env var for CSI migration

### CODE REVIEW

File: pkg/operator/configobserver/storage/storage_operator.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	}()
+
+	storageLister := genericListers.(StorageLister)
+
+	errs := []error{}
+	observedConfig := map[string]interface{}{}
+	storageOperator, err := storageLister.StorageLister().Get(storageOperatorName)
+	if errors.IsNotFound(err) {
+		recorder.Warningf("ObserveStorageOperator", "storage.%s/%s not found", opv1.GroupName, storageOperatorName)
+		return observedConfig, errs
```
Comment (jsafrane) [inline pkg/operator/configobserver/storage/storage_operator.go:47]: I'd feel slightly better with `return observedConfig, nil`

Comment (jsafrane) [reply pkg/operator/configobserver/storage/storage_operator.go:47]: ack

**Comment by dobsonj** [comment]
/assign @deads2k 
for approval

File: pkg/operator/configobserver/storage/storage_operator.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	}
+	if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	newStorageMap := storageToMap(storageOperator)
+	if newStorageMap != nil {
```
Comment (deads2k) [inline pkg/operator/configobserver/storage/storage_operator.go:54]: prefer `len(newStorageMap) > 0`, that works on both nil and empty

File: pkg/operator/configobserver/storage/storage_operator.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	if storage.Spec.VSphereStorageDriver == opv1.CSIWithMigrationDriver {
+		storageMap["OPENSHIFT_DO_VSPHERE_MIGRATION"] = "true"
+	}
+
+	if len(storageMap) == 0 {
+		return nil
+	}
+
+	return storageMap
```
Comment (deads2k) [inline pkg/operator/configobserver/storage/storage_operator.go:84]: nit: equivalent to?

```suggestion
	if storage.Spec.VSphereStorageDriver == opv1.CSIWithMigrationDriver {
	    return map[string]string{"OPENSHIFT_DO_VSPHERE_MIGRATION": "true"}
	}

	return nil
```

**Comment by deads2k** [comment]
minor comments.  I'm probably wiling to forgo them on a 4.13-only change. 

/approve
/hold

hold in case you'd like to update the comments.  I'm ok if you'd like to keep the existing and uhold.

**Comment by dobsonj** [comment]
> hold in case you'd like to update the comments. I'm ok if you'd like to keep the existing and uhold.

Thanks, I accepted the changes, it's a good simplification.

**Pull request merged**